vocabulary-draft/continuous-delivery.md

```diff
@@ -0,0 +1,114 @@
+- **Component Display Name**: UX/human-friendly display name of component, e.g. “C37 Backbone Router”.
+- **Artifacts**: List of artifacts (files, packages, Docker images etc.) part of this component version.
+- **Component Track**: States what track (e.g. platform variant (x86_64, ARM etc.), project (e.g. X-DataCenter, X-Cloud etc.) or other) the component, of which this is a version, belongs to.
+
+### Product Composition Created
+
+Announces that a new composition of source-files, artifacts, component versions and other compositions/baseline versions, has been created.
+
+__Note:__ This event description suffers from a missing piece of terminology to represent the “recipe” by that controls what is included in new compositions. E.g. “The latest stable version (LSV) of this application, the LSV of this library, and the LSV of this configuration file”. For now, this will be referred to as a “Recipe” until a better name is given. (ESCI calls this a Baseline, Eiffel likely has a term for this as well).
+
+Product Composition Created Events MUST include the following attributes:
+
+- **Composition ID**: Unique identifier of composition.
+- **Recipe ID**: Identifier of recipe of which this is a composition.
+- **Content**: List of sources, artifacts, components and other compositions part of this composition
+
+Optional attributes:
+
+- **Version Name**: Machine-friendly and human-friendly version tag/number/etc. for composition, e.g. “1.4.2-alpha2”.
+- **Recipe Name**: Machine-friendly and human-friendly name of recipe, e.g. “gemeni-backbone-gen2”.
+- **Recipe Display Name**: UX/human-friendly display name of recipe, e.g. "Gemini Backbone, Gen2”.
+- **Artifacts**: List of artifacts (files, packages, Docker images etc.) part of this component version.
+- **Component Track**: States what track (e.g. platform variant (x86_64, ARM etc.), project (e.g. X-DataCenter, X-Cloud etc.) or other) the component, of which this is a version, belongs to.
+
+### Confidence Label Verdict
+
+Announces that a Confidence Label Verdict (a.k.a. Confidence Level) for one or more source changes, artifacts, component versions or compositions has been set or updated.
+
+Confidence Label Verdict Events MUST include the following attributes:
+
+- **Confidence Label Verdict ID**: Unique identifier of confidence label verdict.
+- **Name**: Machine-friendly and human-friendly name of confidence label
+- **Verdict**: The verdict of this confidence label result (Success, Failure, Inconclusive)
+- **Targets**: List of sources, artifacts, component versions and compositions this confidence label verdict applies to
+
+Optional attributes:
+
+- **Causes**: List of e.g. pipeline runs or test suites that lead to this confidence label verdict.
+- **Sub Confidence Label Verdicts**: List of other confidence label verdicts that resulted in this confidence label verdict. Used for “summarizing” (e.g. “parent”) confidence labels.
+
+### Environment Allocation Status Changed
+
```

Comment:
@erkist This one makes a lot of sense.. so it is about the status of a specific environment right? Like Test Environment was shut down or it is not available right now? The term allocation sounds too hardware related, in the sense that it makes me think about physical resources not being there for the environment to run. From a developer perspective, I will be more interested to know if the environment is there or not. Possible scenarios that might be interesting to cover: 
- The environment is out of resources (too many services running, there is no more CPU or memory available to run more things) 
- The administrator decided to provision a new Testing environment, hence the old one is removed. 

@erkist Was that the intention of your suggestion? Are there any specifics related to hardware projects that you would like to have for environments?

Reply:
@salaboy you bring up a very important point w.r.t. the "out of resources" comment. Most of the environments in my corner of the domain can only support one "user" at a time, so it has a very simplistic view on the amount of resources.

To explain a bit more what I am after:
1. Every activity or test that needs to run may declare what it needs from the environment (ESCI calls this a [Resource Set Capability](https://github.com/erkist/sig-interoperability/tree/erkist-esci/workstreams/events_in_cicd/input/enterprise-scale-ci#resource-set), Eiffel calls this a [Constraint](https://github.com/eiffel-community/eiffel/blob/master/eiffel-vocabulary/EiffelTestExecutionRecipeCollectionCreatedEvent.md#databatchesrecipesconstraints) )
2. When this activity or test needs to be run, we need to find a suitable environment to run this activity/test on. This is referred to above as Environment Allocation.
3a. If an environment with the right capabilities exists and is currently idle, an event is immediately sent out by the environment allocation service stating "This environment is now allocated to this activity", and an orchestration server can start running.
3b. If an environment with the right capabilities exists but is currently allocated to something else, the orchestration server just waits for the event from 3a (note that other activities/tests may wait for the same environment and get to run first)
3c. If no environment with the right capabilities exist, the actor trying to put the request in the environment allocation queue is immediately informed that the request cannot be fulfilled, and the pipeline is aborted.

I think your comments about using these events for monitoring is also very interesting, but for me this is mainly an orchestration event to make things happen.